> Do you know what use-cases this will speed up? Last time I checked, the URL helpers are often called so deferring the urllib.request import is unlikely to be helpful end-to-end for most commands. Is it just --help and autocomplete?

Adding the `urllib.request` to the existing network import test on main results in these failures

```
FAILED tests/functional/test_cli.py::test_no_network_imports[check] - assert not True
FAILED tests/functional/test_cli.py::test_no_network_imports[debug] - assert not True
FAILED tests/functional/test_cli.py::test_no_network_imports[freeze] - assert not True
FAILED tests/functional/test_cli.py::test_no_network_imports[inspect] - assert not True
FAILED tests/functional/test_cli.py::test_no_network_imports[list] - assert not True
FAILED tests/functional/test_cli.py::test_no_network_imports[show] - assert not True
FAILED tests/functional/test_cli.py::test_no_network_imports[uninstall] - assert not True
``` 

So all of these commands should generally see improvement, but it might depend on the exact invocation / environment. For instance for `pip show`,  `urllib.request` will get imported for editable installs.

--- a/src/pip/_internal/req/req_install.py
+++ b/src/pip/_internal/req/req_install.py
@@ -774,6 +774,11 @@ class InstallRequirement:
         use_user_site: bool = False,
         pycompile: bool = True,
     ) -> None:
+        # Lazy import to avoid transitively importing `_vendor.distlib.compat`
+        # which in turn imports `urllib.request` which is slow.
+        # During an actual installation, `urllib.request` will end up imported anyway,
+        # but `req.req_install` (this module) is also imported from commands that
+        # don't actually install anything (e.g. `pip freeze` or `pip show`).
         from pip._internal.operations.install.wheel import install_wheel
 
         assert self.req is not None

